lib/ui/painting/paint.cc

```diff
@@ -197,20 +144,14 @@ Paint::Paint(Dart_Handle paint_objects, Dart_Handle paint_data) {
   if (filter_quality)
     paint_.setFilterQuality(static_cast<SkFilterQuality>(filter_quality));
 
-  if (uint_data[kColorFilterIndex] && uint_data[kInvertColorIndex]) {
-    sk_sp<SkColorFilter> color_filter = ExtractColorFilter(uint_data, values);
-    if (color_filter) {
-      sk_sp<SkColorFilter> invert_filter =
-          MakeColorMatrixFilter255(invert_colors);
-      paint_.setColorFilter(invert_filter->makeComposed(color_filter));
-    }
-  } else if (uint_data[kInvertColorIndex]) {
-    paint_.setColorFilter(MakeColorMatrixFilter255(invert_colors));
-  } else if (uint_data[kColorFilterIndex]) {
-    sk_sp<SkColorFilter> color_filter = ExtractColorFilter(uint_data, values);
-    if (color_filter) {
-      paint_.setColorFilter(color_filter);
+  if (uint_data[kInvertColorIndex]) {
```

Comment:
This simplification looks great! It would be nice to have a unit test (maybe in a newly created `paint_unittests.cc`) to cover all the cases.